admin/app/controllers/solidus_admin/auth.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module SolidusAdmin::Auth
+  extend ActiveSupport::Concern
+
+  included do
+    if ::SolidusAdmin::Config.authentication_adapter
+      include ::SolidusAdmin::Config.authentication_adapter.constantize
+    end
+
+    before_action :authenticate_solidus_admin_user!
+
+    helper_method :current_solidus_admin_user
+    helper_method :solidus_admin_logout_path
+    helper_method :solidus_admin_logout_method
+  end
+
+  private
+
+  def authenticate_solidus_admin_user!
+    send SolidusAdmin::Config.authentication_method if SolidusAdmin::Config.authentication_method
+  end
```

Comment:
Why not define those methods in the included module adapter?

Reply:
This file contains the adapter mechanics, providing the hooks that different adapter can use. The backend one is of course the one we need while migrating out of it, but eventually I think other will pop up (devise, maybe 3rd party single sign on, etc.) according to "the plan" in the description.